plugins/inputs/slurm/slurm.go

```diff
@@ -0,0 +1,508 @@
+		}
+		if strPtr, ok = jobs[i].GetStandardErrorOk(); ok {
+			records["standard_error"] = strings.ReplaceAll(*strPtr, "\\", "")
+		}
+		if strPtr, ok = jobs[i].GetStandardInputOk(); ok {
+			records["standard_input"] = strings.ReplaceAll(*strPtr, "\\", "")
+		}
+		if strPtr, ok = jobs[i].GetCurrentWorkingDirectoryOk(); ok {
+			records["current_working_directory"] = strings.ReplaceAll(
+				*strPtr, "\\", "")
+		}
+		if int64Ptr, ok = jobs[i].GetSubmitTimeOk(); ok {
+			records["submit_time"] = *int64Ptr
+		}
+		if int64Ptr, ok = jobs[i].GetStartTimeOk(); ok {
+			records["start_time"] = *int64Ptr
+		}
+		if int32Ptr, ok := jobs[i].GetCpusOk(); ok {
+			records["cpus"] = *int32Ptr
+		}
+		if int32Ptr, ok = jobs[i].GetTasksOk(); ok {
+			records["tasks"] = *int32Ptr
+		}
+		if int64Ptr, ok = jobs[i].GetTimeLimitOk(); ok {
+			records["time_limit"] = *int64Ptr
+		}
+		if strPtr, ok = jobs[i].GetTresReqStrOk(); ok {
+			for k, v := range s.parseTres(*strPtr) {
+				records["tres_"+k] = v
+			}
+		}
+
+		acc.AddFields("slurm_jobs", records, tags)
+	}
+}
+
+func (s *Slurm) gatherNodesMetrics(acc telegraf.Accumulator,
+	nodes []goslurm.V0038Node) {
+	var (
+		int32Ptr *int32
+		int64Ptr *int64
+		strPtr   *string
+		ok       bool
+	)
+
+	for _, node := range nodes {
+		records := make(map[string]interface{})
+		tags := make(map[string]string)
+
+		tags["source"] = s.baseURL.Hostname()
+		if strPtr, ok = node.GetNameOk(); ok {
+			tags["name"] = *strPtr
+		}
+
+		if strPtr, ok = node.GetStateOk(); ok {
+			records["state"] = *strPtr
+		}
+		if int32Ptr, ok = node.GetCoresOk(); ok {
+			records["cores"] = *int32Ptr
+		}
+		if int32Ptr, ok = node.GetCpusOk(); ok {
+			records["cpus"] = *int32Ptr
+		}
+		if int64Ptr, ok = node.GetCpuLoadOk(); ok {
+			records["cpu_load"] = *int64Ptr
+		}
+		if int64Ptr, ok = node.GetAllocCpusOk(); ok {
+			records["alloc_cpu"] = *int64Ptr
+		}
+		if int32Ptr, ok = node.GetRealMemoryOk(); ok {
+			records["real_memory"] = *int32Ptr
+		}
+		if int32Ptr, ok = node.GetFreeMemoryOk(); ok {
+			records["free_memory"] = *int32Ptr
+		}
+		if int64Ptr, ok = node.GetAllocMemoryOk(); ok {
+			records["alloc_memory"] = *int64Ptr
+		}
+		if strPtr, ok = node.GetTresOk(); ok {
+			for k, v := range s.parseTres(*strPtr) {
+				records["tres_"+k] = v
+			}
+		}
+		if strPtr, ok = node.GetTresUsedOk(); ok {
+			for k, v := range s.parseTres(*strPtr) {
+				records["tres_used_"+k] = v
+			}
+		}
+		if int32Ptr, ok = node.GetWeightOk(); ok {
+			records["weight"] = *int32Ptr
+		}
+		if strPtr, ok = node.GetSlurmdVersionOk(); ok {
+			records["slurmd_version"] = *strPtr
+		}
+		if strPtr, ok = node.GetArchitectureOk(); ok {
+			records["architecture"] = *strPtr
+		}
+
+		acc.AddFields("slurm_nodes", records, tags)
+	}
+}
+
+func (s *Slurm) gatherPartitionsMetrics(acc telegraf.Accumulator,
+	partitions []goslurm.V0038Partition) {
+	var (
+		int32Ptr *int32
+		strPtr   *string
+		ok       bool
+	)
+
```

Comment:
And once more.

Reply:
Whops! Done on 47393078166c438b5a0f2263661b1170b5c02f67.